packages/block-editor/src/components/block-list/use-in-between-inserter.js

```diff
@@ -159,15 +159,23 @@ export function useInBetweenInserter() {
 					return;
 				}
 
+				if ( event.type === 'click' ) {
+					hideInsertionPoint();
+					selectBlock( clientId, -1 );
+					return;
+				}
+
 				showInsertionPoint( rootClientId, index, {
 					__unstableWithInserter: true,
 				} );
 			}
 
 			node.addEventListener( 'mousemove', onMouseMove );
+			node.addEventListener( 'click', onMouseMove );
 
 			return () => {
 				node.removeEventListener( 'mousemove', onMouseMove );
+				node.removeEventListener( 'click', onMouseMove );
```

Comment:
This onClick handler was previously in the popover, but we can handled it here since it's now click-through.